hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureParameterImpl.java

```diff
@@ -11,19 +11,32 @@
 import java.util.Objects;
 import jakarta.persistence.ParameterMode;
 
-import org.hibernate.NotYetImplementedFor6Exception;
+import org.hibernate.engine.jdbc.cursor.spi.RefCursorSupport;
+import org.hibernate.engine.jdbc.env.spi.ExtractedDatabaseMetaData;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
 import org.hibernate.metamodel.model.domain.AllowableParameterType;
+import org.hibernate.metamodel.model.domain.AllowableTemporalParameterType;
 import org.hibernate.procedure.spi.NamedCallableQueryMemento;
+import org.hibernate.procedure.spi.ParameterStrategy;
 import org.hibernate.procedure.spi.ProcedureCallImplementor;
 import org.hibernate.procedure.spi.ProcedureParameterImplementor;
 import org.hibernate.query.AbstractQueryParameter;
+import org.hibernate.query.spi.QueryParameterBinding;
+import org.hibernate.type.ProcedureParameterNamedBinder;
+import org.hibernate.type.descriptor.ValueBinder;
+import org.hibernate.type.descriptor.jdbc.JdbcTypeDescriptor;
+import org.hibernate.type.spi.TypeConfiguration;
+
+import org.jboss.logging.Logger;
 
 /**
  * @author Steve Ebersole
  */
 public class ProcedureParameterImpl<T> extends AbstractQueryParameter<T> implements ProcedureParameterImplementor<T> {
+	private static final Logger log = Logger.getLogger( ProcedureParameterImpl.class );
+
 	private final String name;
-	private final Integer position;
+	private Integer position;
```

Comment:
Please make this final again.

Reply:
If position is null but named parameter are not supported then I set it while registering the parameters